### DISCUSSION TITLE
Why do We create HelperConfig contracts?

### DISCUSSION THREAD

**Discussion by johan784** [Cyfrin/foundry-full-course-cu#3050] [Q&A] [answered]
Ive understood why deploy scripts are needed, however I dont get why we need to create HelperConfig contracts...

**Comment by EngrPips** [comment, marked as answer]
Hello @johan784, We needed a `HelperConfig` to have a flexible deployment process, which means we could easily deploy our set of contracts to any blockchain of our choice without needing to re-write our deployment script. Switching the value we pass to the `-rpc-url` argument will make our deployment run smoothly on the preferred blockchain. Please ask questions if anything is confusing.

**Reply by johan784** [reply]
thanks!

**Reply by EngrPips** [reply]
I am glad you made sense of the explanation. Keep crushing it.
